libpromises/var_expressions.c

```diff
@@ -95,6 +96,13 @@ static size_t VarRefHash(const VarRef *ref)
     return (h & (INT_MAX - 1));
 }
 
+unsigned int VarRefHash_untyped(const void *ref,
+                                unsigned int seed ARG_UNUSED,
+                                unsigned int max)
+{
+    return VarRefHash(ref) & (max - 1);
+}
```

Comment:
I had one customer with over 100,000 variables (array elements) before I converted them to data containers.